app/controllers/UploadController.scala

```diff
@@ -29,28 +25,27 @@ class UploadController(override val authActions: HMACAuthActions, awsConfig: AWS
 
   private val credsGenerator = new CredentialsGenerator(awsConfig)
 
-  def list(atomId: String) = APIAuthAction {
+  def list(atomId: String) = APIAuthAction { req =>
     val atom = MediaAtom.fromThrift(getPreviewAtom(atomId))
-    val added = ClientAsset.byVersion(atom.assets, youTube.getProcessingStatus)
+    val assets = ClientAsset.fromAssets(atom.assets).map(addYouTubeStatus).map(addMetadata(atom.id, _))
 
-    val runningJobs = stepFunctions.getJobs(atomId)
-    val running = runningJobs.flatMap(getRunning)
+    val jobs = stepFunctions.getJobs(atomId)
+    val uploads = jobs.flatMap(getRunning(assets, _))
 
-    val assets = running ++ added
-    Ok(Json.toJson(assets))
+    Ok(Json.toJson(uploads ++ assets))
   }
 
   def create = LookupPermissions { implicit raw =>
-    parse(raw) { req: CreateRequest =>
+    parse(raw) { req: UploadRequest =>
       if(req.selfHost && !raw.permissions.addSelfHostedAsset) {
         Unauthorized(s"User ${raw.user.email} is not authorised with permissions to upload self-hosted asset")
       } else {
         log.info(s"Request for upload under atom ${req.atomId}. filename=${req.filename}. size=${req.size}, selfHosted=${req.selfHost}")
 
         val atom = MediaAtom.fromThrift(getPreviewAtom(req.atomId))
-        val upload = buildUpload(atom, raw.user, req.size, req.selfHost, req.syncWithPluto)
+        val version = if(atom.assets.isEmpty) { 1 } else { atom.assets.map(_.version).max + 1 }
```

Comment:
minor: would be nice if there was a `getNextVersion` helper function on `MediaAtom` for this